src/main/java/org/apache/maven/buildcache/BuildCacheMojosExecutionStrategy.java

```diff
@@ -134,7 +134,7 @@ public void execute(
             if (restorable) {
                 CacheRestorationStatus cacheRestorationStatus =
                         restoreProject(result, mojoExecutions, mojoExecutionRunner, cacheConfig);
-                restored &= CacheRestorationStatus.SUCCESS == cacheRestorationStatus;
+                restored = CacheRestorationStatus.SUCCESS == cacheRestorationStatus;
```

Comment:
The `restored` variable name is misleading since it has 2 meanings:
1) It starts as meaning _The build is fully cached_.
2) The meaning may then change to _The build was restored from cache_.

I suggest changing
`boolean restored = result.isSuccess()`
to
`boolean restored = false`

To improve that.

Reply:
Thanks for the feedback on the PR! I agree that the dual meaning of `restored` is confusing. I would take your suggestion to improve the code.